example/ffi/ndt.cpp

```diff
@@ -0,0 +1,46 @@
+/* Public domain 2017, Simone Basso <bassosimone@gmail.com. */
+
+#include <measurement_kit/ffi.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/*
+ * Ideally it would be nice to have this written in C. For this to work,
+ * however, we would need to specifically compile Measurement Kit to link
+ * statically with its own version of libc++.
+ */
+int main() {
+    char buff[128];
+    mk_task_t *task = mk_task_start_ex("{\n"
+                                       "    \"type\": \"Ndt\",\n"
+                                       "    \"verbosity\": \"INFO\"\n"
+                                       "}\n",
+            buff, sizeof(buff));
+    if (task == NULL) {
+        fprintf(stderr, "ERROR: cannot create/start task: %s\n", buff);
+        exit(1);
+    }
+
+    int done = 0;
+    do {
+        mk_event_t *event = mk_task_wait_for_next_event(task);
+        if (event == NULL) {
+            fprintf(stderr, "ERROR: cannot wait for next event\n");
+            exit(1);
+        }
+        const char *serio = mk_event_serialize(event);
+        if (serio == NULL) {
+            fprintf(stderr, "ERROR: cannot serialize event\n");
+            exit(1);
+        }
+        // A bit ugly but this is `null` as a serialized JSON
+        done = (strcmp(serio, "null") == 0);
```

Comment:
It seems somewhat awkward to have the task completion come out as a not-explicit event on the event stream.

How hard would it be to implement a `mk_task_is_active(task)` or equivalent that could be used as a condition in a while loop?

Reply:
Not hard. It was part of a previous draft. I think I omitted it because I was going down the path of reducing complexity recursively and probably exaggerated a bit :-).